pkg/providers/ovirt/client/ovirt-client.go

```diff
@@ -0,0 +1,270 @@
+package ovirtclient
+
+import (
+	"fmt"
+
+	ovirtsdk "github.com/ovirt/go-ovirt"
+)
+
+// OvirtClient retrieves rich VM object - with all necessary links have been followed
+type OvirtClient interface {
+	GetVM(connectionSettings *ConnectionSettings, id *string, name *string, cluster *string) (*ovirtsdk.Vm, error)
+}
+
+// ConnectionSettings wrap information required to make oVirt API connection
+type ConnectionSettings struct {
+	URL      string
+	Username string
+	Password string
+	Insecure bool
+}
+
+// RichOvirtClient is responsilbe for retrieving VM data from oVirt API
+type RichOvirtClient struct {
+}
+
+// GetVM rertrieves oVirt VM data for given id or name and cluster. VM will have certain links followed and updated.
+func (client *RichOvirtClient) GetVM(cs *ConnectionSettings, id *string, name *string, cluster *string) (*ovirtsdk.Vm, error) {
+	con, err := client.createConnection(cs.URL, cs.Username, cs.Password, cs.Insecure)
```

Comment:
please add defer con.Close(). We do not want to leak open connections.

Reply:
:+1: